### PR TITLE
Fix typos in comments and Kconfig files

### PR DESCRIPTION
**Pull request by juniskane**
## Summary
Just some spelling errors and other minor corrections to comments here and there, accumulated over time.

## Impact
No functional change.

## Testing
N7A

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

File: sched/Kconfig [hunk 1]
```diff
@@ -13,14 +13,6 @@ menuconfig DISABLE_OS_API
...
 		of those APIs.
 
-		There are certain dependency relationships in these
-		features.
-
-		1) mq_notify logic depends on signals to awaken tasks
-		waiting for queues to become full or empty.
-		2) pthread_condtimedwait() depends on signals to wake
-		up waiting tasks.
-
```
Comment (hartmannathan) [inline sched/Kconfig:23]: Everything else looks good, but why remove this section? Is it stale?

**Review by hartmannathan — APPROVED** [review]
LGTM

File: arch/Kconfig [hunk 2]
```diff
@@ -423,7 +423,7 @@ config ARCH_USE_TEXT_HEAP
 	default n
 	depends on ARCH_HAVE_TEXT_HEAP
 	---help---
-		This option enables architecture-sepecific memory allocator
+		This option enables architecture-specific memory allocator
 		for dynamic code loading. For example, ESP32 has a separate memory
```
Comment (gustavonihei) [inline arch/Kconfig:427]:
```suggestion
		for dynamic code loading. For example, ESP32 has separate memory
```

File: sched/Kconfig (hunk 1 above)
Comment (juniskane) [reply sched/Kconfig:23]: That section was meaningful only when we had CONFIG_DISABLE_SIGNALS. Makes little sense to list dependencies to features that cannot in fact be removed. Wording "in these features" refers to the set of features than can be disabled by subitems of CONFIG_DISABLE_OS_API so it is misleading to imply that signals are part of it.

**Pull request merged**